crates/polars-core/src/chunked_array/comparison/mod.rs

```diff
@@ -968,28 +968,52 @@ impl Not for BooleanChunked {
 }
 
 impl BooleanChunked {
-    /// Check if all values are `true`
+    /// Returns whether all values in the column are `true`.
     pub fn all(&self) -> bool {
         self.downcast_iter().all(compute::boolean::all)
     }
 
-    /// Check if any value is `true`
+    /// Returns whether any value in the column is `true`.
     pub fn any(&self) -> bool {
         self.downcast_iter().any(compute::boolean::any)
     }
 
-    // Three-valued versions which can return None
-    pub fn all_3val(&self, drop_nulls: bool) -> Option<bool> {
-        if drop_nulls || self.null_count() == 0 {
-            Some(self.all())
+    /// Returns whether all values in the column are `true`.
+    ///
+    /// If `ignore_nulls` is set to `false`, returns `None` if the array
+    /// contains any null values and no `false` values.
+    pub fn all_ternary(&self, ignore_nulls: bool) -> Option<bool> {
+        if ignore_nulls || self.null_count() == 0 {
+            return Some(self.all());
+        }
+
+        // Inverse version of `any` - check if any values are False
+        let contains_false = self.downcast_iter().any(|array| {
+            if array.is_empty() {
+                false
```

Comment:
If it's empty the null count will be 0, so this is handled by the case above.

Reply:
This closure operates on array chunks, which may be empty / have no validity mask.